contracts/MiniMeToken.sol

```diff
@@ -262,20 +262,15 @@ contract MiniMeToken is ERC20, Controlled {
     function balanceOfAt(address _owner, uint _blockNumber) constant
         returns (uint) {
 
-        // If the `_blockNumber` requested is before the genesis block for the
-        //  the token being queried, the value returned is 0
-        if (_blockNumber < creationBlock) {
-            return 0;
-
         // These next few lines are used when the balance of the token is
         //  requested before a check point was ever created for this token, it
         //  requires that the `parentToken.balanceOfAt` be queried at the
         //  genesis block for that token as this contains initial balance of
         //  this token
-        } else if ((balances[_owner].length == 0)
+        if ((balances[_owner].length == 0)
             || (balances[_owner][0].fromBlock > _blockNumber)) {
             if (address(parentToken) != 0) {
-                return parentToken.balanceOfAt(_owner, parentSnapShotBlock);
+                return parentToken.balanceOfAt(_owner, min(_blockNumber, parentSnapShotBlock));
```

Comment:
This is almost surely fine *unless* there is any reason to expect that the chain of parent tokens will every become sufficiently long that this recursive calling up the parent chain will exceed the block gas limit.